models/cms_hcc/final/cms_hcc__patient_risk_scores_monthly.sql

```diff
@@ -0,0 +1,225 @@
+        , model_version
+        , payment_year
+        , collection_start_date
+        , collection_end_date
+
+)
+
+/*
+    CMS Guidance for the transition from v24 to v28:
+
+    PY2024 risk scores will be blended using 67% of the risk score calculated
+    from v24 and 33% from v28.
+
+    PY2025 risk scores will be blended using 33% of the risk score calculated
+    from v24 and 67% from v28.
+
+    PY2026 risk scores will be 100% from v28.
+
+    Prior payment years will still be calculated from v24 only.
+*/
+, transition_scores as (
+
+    select
+          patient_id
+        , risk_score
+        , case
+            when payment_year <= 2023 and model_version = 'CMS-HCC-V24' then risk_score
+            when payment_year = 2024 and model_version = 'CMS-HCC-V24' then risk_score * 0.67
+            when payment_year = 2025 and model_version = 'CMS-HCC-V24' then risk_score * 0.33
+            when payment_year >= 2026 and model_version = 'CMS-HCC-V24' then 0
+            end as v24_risk_score
+        , case
+            when payment_year <= 2023 and model_version = 'CMS-HCC-V28' then 0
+            when payment_year = 2024 and model_version = 'CMS-HCC-V28' then risk_score * 0.33
+            when payment_year = 2025 and model_version = 'CMS-HCC-V28' then risk_score * 0.67
+            when payment_year >= 2026 and model_version = 'CMS-HCC-V28' then risk_score
+            end as v28_risk_score
+        , model_version
+        , payment_year
+        , collection_start_date
+        , collection_end_date
+    from raw_score
+
+)
+
+/*
+    Grouping by patient to create a single row per patient.
+*/
+, transition_scores_grouped as (
+
+    select
+          patient_id
+        , max(v24_risk_score) as v24_risk_score
+        , max(v28_risk_score) as v28_risk_score
+        , payment_year
+        , collection_start_date
+        , collection_end_date
+    from transition_scores
+    group by
+          patient_id
+        , payment_year
+        , collection_start_date
+        , collection_end_date
+
+)
+
+, blended as (
+
+    select
+          patient_id
+        , v24_risk_score
+        , v28_risk_score
+        , v24_risk_score + v28_risk_score as blended_risk_score
+        , payment_year
+        , collection_start_date
+        , collection_end_date
+    from transition_scores_grouped
+
+)
+
+, normalized as (
+
+    select
+          blended.patient_id
+        , blended.v24_risk_score
+        , blended.v28_risk_score
+        , blended.blended_risk_score
+        , blended.blended_risk_score / seed_adjustment_rates.normalization_factor as normalized_risk_score
+        , blended.payment_year
+        , blended.collection_start_date
+        , blended.collection_end_date
+    from blended
+        left join seed_adjustment_rates
+            on least(
+                blended.payment_year, 
+                (select max(payment_year) from seed_adjustment_rates)
+            ) = seed_adjustment_rates.payment_year
```

Comment:
I think that would be ok, if the rates are available and added reasonably early in the year. The point would be to track progress for the current collection year, but some idea of progress could be tracked with the blended score before the adjustment rate is available. Would we be able to get PY2025 added to the seed?

Reply:
Just to point it out, using the previous year's value is also an approximation like using blended score, so it would definitely be better to use the actual rate when it's available 😃  rather than the version we have implemented here. I also think in terms of generalized use in other organizations, falling back to the previous year like we've done might lead to misunderstood assumptions! All this just to say I agree with your recommendation and hope we can get PY2025 added.

Reply:
@wtyree that totally makes sense to me. I just added the rates for PY2025 to the seed. It will be released in the `v0.12.4` patch.